File: src/minim/controller/table/misc/OneShot.java
package minim.controller.table.misc;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import minim.controller.table.Lines;
import minim.controller.table.Table;
import minim.controller.table.adventure.Type;
import minim.controller.table.adventure.dungeon.Dungeon;
import minim.controller.table.bold.Waylay;
import minim.controller.table.hexcrawl.Hexcrawl;
import minim.controller.table.instant.InstantSetting;
import minim.controller.table.toon.CartoonAdventure;
import minim.controller.table.weather.Calendar;

public class OneShot extends Table {
	static final List<String> ARC = List.of("Setup", "Confrontation", "Resolution");
	static final Map<String, Table> ADVENTURES = new HashMap<>();

	static {
		ADVENTURES.put(Type.DUNGEON, Dungeon.SINGLETON);
		ADVENTURES.put(Type.URBAN, CartoonAdventure.SINGLETON);
		ADVENTURES.put(Type.HEXCRAWL, Hexcrawl.SMALL);
	}

	public OneShot() {
		super("One-shot adventure");
	}

	@Override
	public void build() {
		// don't
	}

	@Override
	public String roll() {
		var l = new Lines();
		l.add("Setting", InstantSetting.SINGLETON);
		for (var a : ARC) {
			l.add();
			l.add(a + " chapter", Waylay.SINGLETON);
			l.add();
			var t = Type.SINGLETON.roll();
			l.add("%s module (%s):".formatted(a, t.toLowerCase()));
			l.add(ADVENTURES.get(t));
		}
		l.add();
		l.add(Calendar.SINGLETON);
		return l.toString();
	}

}

File: web/script/control/table/adventure/oneshot.js
import * as table from '../table.js'
import * as instant from '../instant.js'
import * as weather from '../weather.js'
import * as bold from '../bold.js'
import * as dungeonm from './dungeon.js'
import * as hexcrawl from './hexcrawl.js'
import * as cartoon from './cartoon.js'

export var tables=[]

export var dungeon="Dungeon"
export var wilderness="Wilderness"
export var urban="Urban"

var type=new table.Table('Adventure, one-shot, type',[dungeon,wilderness,urban])
tables.push(type)

const ARC = ["Setup", "Confrontation", "Resolution"]
const ADVENTURES = new Map([
    [dungeon,dungeonm.tables[dungeonm.tables.length-1]],
    [wilderness,hexcrawl.BIG],
    [urban, cartoon.adventure]
])
class OneShot extends table.Table {
  constructor() {
		super("Adventure, one-shot");
	}

  roll() {
		var l = [];
		l.push("Setting: "+instant.setting);
		for (let a of ARC) {
			l.push('');
			l.push(a + " chapter: "+ bold.waylay);
			l.push('');
			var t = type.roll();
			l.push(`${a} module (${t.toLowerCase()}):`)
			l.push(ADVENTURES.get(t));
		}
		l.push();
    l.push(weather.calendar);
		return l.join('<br/>')
	}
}

tables.push(new OneShot())
